AgentDiscoveries-Frontend/app/end_to_end_tests/login_test.js

```diff
@@ -0,0 +1,113 @@
+import chai from 'chai'
+import React from 'react'
+import * as webdriver from 'selenium-webdriver'
+import * as chrome from 'selenium-webdriver/chrome'
+//The next import is required to ensure chromedriver is on the system path
+import * as chromeDriver from 'chromedriver';
+
+let expect = chai.expect;
+
+describe("login-page", () => {
+    "use strict";
+    const url = process.argv[process.argv.length - 1].valueOf();
+    const loginUrl = url + "/#/login";
+    let driver;
+
+    before(function () {
+        const chromeOptions = new chrome.Options();
+        chromeOptions.addArguments('headless', 'disable-gpu');
+        driver = new webdriver.Builder()
+            .forBrowser('chrome')
+            .setChromeOptions(chromeOptions)
+            .build();
+    });
+
+    after(function () {
+        driver.quit();
+    });
+
+
+    it('loads', function (done) {
+        this.timeout(5000);
+        driver.get(loginUrl)
+            .then(() => {
+                return driver.getTitle()
+            })
+            .then(res => {
+                expect(res).to.equal("Agent Discoveries")
+                done();
+            })
+    });
+
+    it('has a submit button', function (done) {
+        this.timeout(5000);
+        driver.get(loginUrl)
+            .then(() => {
+                const element = driver.findElement(webdriver.By.id('login-submit'));
+                driver.wait(webdriver.until.elementIsVisible(element), 5000)
+            })
+            .then(() => {
+                return driver.findElement(webdriver.By.id('login-submit'));
+            })
+            .then(element => {
+                return element.getTagName()
+            })
+            .then(tag => {
+                expect(tag).to.equal('button');
+                done();
+            })
+    });
+
+    it('does not change page on attempt to log in without credentials', function (done) {
+        this.timeout(5000);
+        const element = driver.findElement(webdriver.By.id('login-submit'));
+        driver.get(loginUrl)
+            .then(() => {
+                driver.wait(webdriver.until.elementIsVisible(element), 5000)
+            })
+            .then(() => {
+                element.click()
+            })
+            .then(() => {
+                return driver.getCurrentUrl()
+            })
+            .then(url => {
+                expect(url).to.equal(loginUrl);
+                done();
+            })
+    });
+
+    it('changes page on logging in with credentials', function (done) {
```

Comment:
I'd expect these tests would focus on the login being successful or not - the change of URL feels slightly less important!

Reply:
I wasn't sure how to verify logged in, I guess checking the token is set probably makes more sense!

Reply:
Yep, that's possible - if it's stored somewhere accessible (you can get the webdriver to execute arbitrary JavaScript).

Otherwise, these tests are meant to be from the perspective of a real user - so consider how you would determine whether login was successful if you were using the site...